pkgs/top-level/all-packages.nix

```diff
@@ -40537,6 +40537,19 @@ with pkgs;
 
   nix = nixVersions.stable;
 
+  # The aws-sdk-cpp tests are flaky.  Since pull requests to staging
+  # cause nix to be rebuilt, this means that staging PRs end up
+  # getting false CI failures due to whatever is flaky in the AWS
+  # SDK tests.  Since none of our CI needs to (or should be able to)
+  # contact AWS S3, let's just omit it all from the Nix that runs
+  # CI.  Bonus: the tests build way faster.
+  nixVersions-without-aws-sdk =
+    builtins.mapAttrs (_: pkg:
+      if lib.isDerivation pkg
+      then pkg.override { withAWS = false; }
+      else pkg)
+      nixVersions;
+
```

Comment:
This is a more involved change now though. This adds a new attribute to the effective interface of Nixpkgs that needs some stability guarantees. I think it would be best for now to keep the change local to lib.

Reply:
You're still changing all-packages.nix

Reply:
Thanks for catching that!